pkg/util/circuitbreaker/cirquit_breaker.go

```diff
@@ -0,0 +1,27 @@
+package circuitbreaker
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/sony/gobreaker/v2"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func UnaryClientInterceptor(cb *gobreaker.CircuitBreaker[any]) grpc.UnaryClientInterceptor {
+	return func(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+		_, err := cb.Execute(func() (interface{}, error) {
+			return nil, invoker(ctx, method, req, reply, cc, opts...)
+		})
+		// gobreaker returns unwrapped errors.
+		switch err {
+		case nil:
+		case gobreaker.ErrOpenState,
```

Comment:
nit: should this be checked with error.Is  / errors.As or is it guaranteed the library does not return wrapped errors?

Reply:
Thanks for asking! It is guaranteed that the errors are not wrapped. I was on the fence but finally settled on this. I think if there are questions, it's better to go another way

Reply:
Addressed in [clarify gobreaker error handling](https://github.com/grafana/pyroscope/pull/3505/commits/c31c9c8b679291a0a663f947f8fd1d9054c384f9)